packages/stack/deployment/test/deployment.spec.js

```diff
@@ -0,0 +1,186 @@
+import sinon from 'sinon';
+import assert from 'assert';
+import Deployment from '../src/';
+
+const events = {
+  listeners: {},
+  setCommandHandler: sinon.spy((cmd, fn) => {
+    events.listeners[cmd] = fn;
+  }),
+  request: sinon.spy((cmd, ...args) => {
+    assert(events.listeners[cmd]);
+    events.listeners[cmd](...args);
+  }),
+  emit: (name) => {}
+};
+
+const plugins = {
+  listeners: {},
+  _plugin: {
+    listeners: {},
+    registerActionForEvent: (name, cb) => {
+      if (!plugins._plugin.listeners[name]) {
+        plugins._plugin.listeners[name] = [];
+      }
+      plugins._plugin.listeners[name].push(cb);
+    }
+  },
+  createPlugin: (name, opts) => {
+    return _plugin;
+  },
+  emitAndRunActionsForEvent: sinon.spy((name, options, cb) => {
+    if (plugins._plugin.listeners[name]) {
+      plugins._plugin.listeners[name].forEach(fn => {
+        fn(options, cb);
+      });
+    }
+  })
+};
```

Comment:
This looks a bit hairy, but as commented below, will be moved into a dedicated `embark-testing` package. Notice how we need to reimplement `plugins` and `plugin` to replicate the event system. Looks like another flaw in our architecture. They should be either both exposed on `plugins` or maybe even on `events`.

Thoughts?

Reply:
Totally agree that everything should all be in `plugin` so taht it's exposed in `embark` for plugins.